virtfusion_direct_provisioning.php

```diff
@@ -1027,20 +1027,13 @@ public function editService($package, $service, array $vars = null, $parent_pack
         }
 
         // Return all the service fields
-        $fields = ['virtfusion_server_id', 'virtfusion_hostname', 'virtfusion-os_template', 'virtfusion_password', 'virtfusion-base_ips', 'additional_num_ips', 'virtfusion_ip'];
-        $encrypted_fields = [];
-        $return = [];
-        foreach ($fields as $field) {
-            if (isset($vars[$field]) || isset($service_fields->{$field})) {
-                $return[] = [
-                    'key' => $field,
-                    'value' => $vars[$field] ?? $service_fields->{$field},
-                    'encrypted' => (in_array($field, $encrypted_fields) ? 1 : 0)
-                ];
-            }
+        $fields = [];
+        $encrypted_fields = ['virtfusion_password'];
+        foreach ($service_fields as $key => $value) {
```

Comment:
This may be problematic.  It gets all the current service fields and returns them.  It will ignore updates from $vars, including updated values and added fields.